Documentation refactoring (#221)
sinonjs/lolex: CONTRIBUTING.md

# Contributing to lolex

## Contributing to the code base

### Running tests

Lolex has a comprehensive test suite.
When contributing bug fixes it is important to make sure no tests were broken.
New features also require tests to be added so we can make sure they do not break between versions.

#### On Node.js:

```sh
npm test
```

Or, if you prefer more verbose output:

```
npx mocha ./test/lolex-test.js
```

#### In the browser

> [Mochify](https://github.com/mantoni/mochify.js) is used to run the tests in PhantomJS.

Reviewer: This is no longer correct. Mochify uses headless Chrome now.

Author: @mantoni 
```suggestion
[Mochify](https://github.com/mantoni/mochify.js) is used to run the tests in Headless Chrome.
```

```sh

> npm test-headless

Reviewer 2: Either `npx` or `npm run`

Author: :man_facepalming: sure!
Your choice :smile: 

```suggestion
npx test-headless
```
```suggestion
npm  run test-headless
```

# Contributing to lolex

## Contributing to the code base

### Running tests

Lolex has a comprehensive test suite. If you're thinking of contributing bug
fixes or suggesting new features, you need to make sure you have not broken any

> tests. You are also expected to add tests for any new behavior.

Reviewer 2: "new or changed"?

Author: That makes sense to me, thanks

> #### On node:

Reviewer 3: ```suggestion
#### On Node.js:
```

```sh
npm test
```

Or, if you prefer more verbose output:

```

> $(npm bin)/mocha ./test/lolex-test.js

Reviewer 2: `npx mocha` instead? Less *nix bound

Author: Good choice, we could also add new task to the `package.json` exrcuting `mocha ./test/lolex-test.js`, e.g `test-verbose`?

> $(npm bin)/mocha ./test/lolex-test.js

Reviewer 3: ```suggestion
npx mocha ./test/lolex-test.js
```

```

#### In the browser

[Mochify](https://github.com/mantoni/mochify.js) is used to run the tests in

> PhantomJS. Make sure you have `phantomjs` installed. Then:

Reviewer 3: Doesn't `npm install` already take care of that?

Reviewer 3: If this is a lot of work and requires a fresh install - would you like me to check this?

Author: Hey sorry I left this unanswered... I'm not sure here, would be nice if you can have look though :+1:

Reviewer 3: Verified, we can remove "Make sure you have `phantomjs` installed - `npm install` will install it.
